nativeauthenticator/handlers.py

```diff
@@ -43,33 +43,35 @@ async def get(self):
         )
         self.finish(html)
 
-    def get_result_message(self, user, username):
+    def get_result_message(self, user, taken):
         alert = 'alert-info'
-        message = 'Your information have been sent to the admin'
+        message = 'Your information has been sent to the admin'
 
-        if self.authenticator.open_signup:
-            alert = 'alert-success'
-            message = ('The signup was successful. You can now go to '
-                       'home page and log in the system')
-        if not user:
+        # Always error if username is taken.
+        if taken:
             alert = 'alert-danger'
-            pw_len = self.authenticator.minimum_password_length
-            taken = self.authenticator.user_exists(username)
```

Comment:
This is the line that caused the errors. "username" here is a boolean indicating wether the name exists in the system, but it is used like a string containing a username. Apart from this being nonsensical, during jupyterhub username normalisation, `.lower()` is called on this variable, which throws an error because that's not defined on booleans, leading to a 500 error.

Reply:
that's a great catch